crates/astria-composer/src/metrics_init.rs

```diff
@@ -0,0 +1,234 @@
+//! Crate-specific metrics functionality.
+//!
+//! Registers metrics & lists constants to be used as metric names throughout crate.
+
+use metrics::{
+    describe_counter,
+    describe_gauge,
+    describe_histogram,
+    Unit,
+};
+
+/// Labels
+pub(crate) const ROLLUP_ID_LABEL: &str = "rollup_id";
+
+/// Registers all metrics used by this crate.
+#[allow(clippy::too_many_lines)]
+pub fn register() {
+    // geth collectors metrics
+    describe_counter!(
+        GETH_COLLECTOR_TRANSACTIONS_COLLECTED,
+        Unit::Count,
+        "The number of transactions received by the geth collector labelled by rollup"
+    );
+    describe_counter!(
+        GETH_COLLECTOR_TRANSACTIONS_DROPPED,
+        Unit::Count,
+        "The number of transactions dropped by the geth collector before bundling it labelled by \
+         rollup"
+    );
+    describe_counter!(
+        GETH_COLLECTOR_TRANSACTIONS_FORWARDED,
+        Unit::Count,
+        "The number of transactions successfully sent by the geth collector to be bundled \
+         labelled by rollup"
+    );
+    describe_histogram!(
+        GETH_COLLECTOR_CONNECTION_LATENCY,
+        Unit::Milliseconds,
+        "The time taken to connect to geth"
+    );
+
+    // grpc collector metrics
+    describe_counter!(
+        GRPC_COLLECTOR_TRANSACTIONS_COLLECTED,
+        Unit::Count,
+        "The number of transactions received by the grpc collector"
+    );
+    describe_counter!(
+        GRPC_COLLECTOR_TRANSACTIONS_DROPPED,
+        Unit::Count,
+        "The number of transactions dropped by the grpc collector before sending it to be bundled"
+    );
+    describe_counter!(
+        GRPC_COLLECTOR_TRANSACTIONS_FORWARDED,
+        Unit::Count,
+        "The number of transactions successfully sent by the grpc collector before sending it to \
+         be bundled"
+    );
+
+    // executor metrics
+    describe_counter!(
+        TRANSACTIONS_RECEIVED,
+        Unit::Count,
+        "The number of transactions successfully received from collectors and bundled labelled by \
+         rollup"
+    );
+    describe_counter!(
+        TRANSACTIONS_DROPPED_TOO_LARGE,
+        Unit::Count,
+        "The number of transactions dropped because they were too large"
+    );
+    describe_counter!(
+        NONCE_FETCH_COUNT,
+        Unit::Count,
+        "The number of times we have attempted to fetch the nonce"
+    );
+    describe_counter!(
+        NONCE_FETCH_FAILURE_COUNT,
+        Unit::Count,
+        "The number of times we have failed to fetch the nonce"
+    );
+    describe_histogram!(
+        NONCE_FETCH_LATENCY,
+        Unit::Milliseconds,
+        "The latency of nonce fetch"
+    );
+    describe_gauge!(CURRENT_NONCE, Unit::Count, "The current nonce");
+    describe_histogram!(
+        TRANSACTION_SUBMISSION_LATENCY,
```

Comment:
I find the distinction between `bundle` and `transaction` in this context very confusing. We should either talk about "bundle (of transcations)" or just "transactions" everywhere.

Reply:
I have done the following:
Modify transaction_submission_latency to sequencer_submission_latency and transaction_failure_count to sequencer_failure_count. With this, the Transaction word is now only reserved for rollup transactions